src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,160 @@
+	const logFlag = "log"
+	const executableFlag = "executable"
+	const argsFlag = "args"
+	const outputFlag = "output"
+	const debugFlag = "debug"
+	const daemonFlag = "daemon"
+	flag.StringVar(&controlDir, controlFlag, "", "working directory")
+	flag.StringVar(&resultPath, resultFlag, "", "full path of the result file")
+	flag.StringVar(&logPath, logFlag, "", "full path of the log file")
+	flag.StringVar(&executable, executableFlag, "", "path to the executable being launched")
+	flag.StringVar(&args, argsFlag, "", "(optional) argument(s) to the executable, separated by commas WITH spaces")
+	flag.StringVar(&outputPath, outputFlag, "", "(optional) if recording output, full path of the output file")
+	flag.BoolVar(&debug, debugFlag, false, "(optional) noisy output to log")
+	flag.BoolVar(&daemon, daemonFlag, false, "(optional) Free binary from parent process")
+	flag.Parse()
+
+	// Validate that the required flags were all command-line defined
+	required := []string{controlFlag, resultFlag, logFlag, executableFlag}
+	defined := make(map[string]string)
+	flag.Visit(func(f *flag.Flag) {
+		defined[f.Name] = f.Value.String()
+	})
+	if !common.ValidateFlags(defined, required) {
+		os.Exit(-2)
+	}
+
+	fmt.Fprintf(os.Stdout, "Parent pid is: %v\n", os.Getppid())
+
+	if daemon {
+		fmt.Fprintf(os.Stdout, "1st launch pid is: %v\n", os.Getpid())
+		rebuiltArgs := common.RebuildArgs(defined, daemonFlag)
+		doubleLaunchCmd := exec.Command(os.Args[0], rebuiltArgs...)
+		doubleLaunchCmd.Stdin = nil
+		doubleLaunchCmd.Stdout = nil
+		doubleLaunchCmd.Stderr = nil
+		var sysAttr windows.SysProcAttr
+		sysAttr.HideWindow = true
+		sysAttr.CreationFlags = windows.CREATE_NEW_PROCESS_GROUP
+		sysAttr.NoInheritHandles = true
+		doubleLaunchCmd.SysProcAttr = &sysAttr
+		doubleLaunchErr := doubleLaunchCmd.Start()
+		if doubleLaunchErr != nil {
+			panic("Double launch failed, exiting")
+		}
+		return
+	}
+	// Prepare logging
+	logFile, logErr := os.Create(logPath)
+	if logErr != nil {
+		fmt.Fprintf(os.Stderr, "Unable to create log file: %s", logErr)
+		common.RecordExit(-2, resultPath, log.Default())
+		return
+	}
+	defer logFile.Close()
+	mainLogger, hbLogger, launchLogger, scriptLogger := common.PrepareLogging(logFile, debug)
+
+	for key, val := range defined {
+		mainLogger.Printf("%v: %v", key, val)
+	}
+	mainLogger.Printf("Main pid is: %v\n", os.Getpid())
+	mainLogger.Printf("Parent pid is: %v\n", os.Getppid())
+
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	go common.SignalCatcher(sigChan, mainLogger)
+
+	exitChan := make(chan bool)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go launcher(&wg, exitChan, executable, args, resultPath, outputPath, launchLogger, scriptLogger)
+	go common.Heartbeat(&wg, exitChan, controlDir, resultPath, logPath, hbLogger)
+	mainLogger.Println("about to wait")
+	wg.Wait()
+	mainLogger.Println("done waiting")
+	signal.Stop(sigChan)
+	close(sigChan)
+	mainLogger.Println("done.")
```

Comment:
Would the stuff sent to `mainLogger` here show up in the Jenkins build log by default?

Reply:
main logger does not show by default. Only scriptlogger does

Reply:
this was for extra debugging though, removed in e87b44bb9fc6c71cebd5be23ec0a757889c35423